Refresh.GameServer/Endpoints/Game/ReportingEndpoints.cs

```diff
@@ -4,19 +4,69 @@
 using Bunkum.Listener.Protocol;
 using Bunkum.Protocols.Http;
 using Refresh.GameServer.Database;
+using Refresh.GameServer.Extensions;
+using Refresh.GameServer.Time;
+using Refresh.GameServer.Types.Levels;
+using Refresh.GameServer.Types.Photos;
 using Refresh.GameServer.Types.Report;
+using Refresh.GameServer.Types.UserData;
 
 namespace Refresh.GameServer.Endpoints.Game; 
 
 public class ReportingEndpoints : EndpointGroup 
 {
     [GameEndpoint("grief", HttpMethods.Post, ContentType.Xml)]
-    public Response UploadReport(RequestContext context, GameDatabaseContext database, GameReport body)
+    public Response UploadReport(RequestContext context, GameDatabaseContext database, GameReport body, GameUser user, IDateTimeProvider time)
     {
-        if ((body.LevelId != 0 && database.GetLevelById(body.LevelId) == null) || body.Players is { Length: > 4 } || body.ScreenElements is { Player.Length: > 4 })
-        {
+        GameLevel? level = database.GetLevelById(body.LevelId);
+
+        //If the level is specified but its invalid, return BadRequest
+        if (body.LevelId != 0 && level == null)
             return BadRequest;
+
+        if (user.RedirectGriefReportsToPhotos)
+        {
+            List<SerializedPhotoSubject> subjects = new();
+            if (body.Players != null)
+                subjects.AddRange(body.Players.Select(player => new SerializedPhotoSubject
+                {
+                    Username = player.Username,
+                    DisplayName = player.Username,
+                    //TODO: im not sure what BoundsList expects, seems to be some kind of float value,
+                    //      but im not sure how to get that from a Rect
+                    BoundsList = "", 
```

Comment:
You can pull a couple examples from the tests

Reply:
I do see how its formatted, but im not sure what the values mean, is it `-1 - 1`? `0 - 1`?

Reply:
0 - 1 i believe

Reply:
As for meaning I'm guessing that it's percents per left/top/bottom/right of the image

Reply:
![image](https://github.com/LittleBigRefresh/Refresh/assets/34501060/6febbe91-d754-4134-9e34-6ccb1b77be03)
![image](https://github.com/LittleBigRefresh/Refresh/assets/34501060/7fd1c299-b44e-4208-b5ed-57068143a1dc)
After checking my database, it has negative values, so im going to assume its a NDC type deal and its `-1 - 1`

Reply:
I'm guessing 0,0,0,0 is center then?

Reply:
player in bottom left
![image](https://github.com/LittleBigRefresh/Refresh/assets/34501060/43fd7b73-5c54-4f47-99d2-62e13ce335b5)
player in top right
![image](https://github.com/LittleBigRefresh/Refresh/assets/34501060/6fe340c4-86a2-45fa-97a8-4d62c7f71881)